rotkehlchen/chain/evm/decoding/curve_lend/decoder.py

```diff
@@ -0,0 +1,617 @@
+
+    def _get_possibly_cached_vault_address(
+            self,
+            cache_type: Literal[
+                CacheType.CURVE_VAULT_CONTROLLER,
+                CacheType.CURVE_VAULT_AMM,
+                CacheType.CURVE_VAULT_BORROWED_TOKEN,
+                CacheType.CURVE_VAULT_COLLATERAL_TOKEN,
+            ],
+            vault_address: 'ChecksumEvmAddress',
+            contract_method: str,
+    ) -> ChecksumEvmAddress | None:
+        """Get address from the cache or the specified vault contract method.
+        Returns the address or None on error."""
+        with GlobalDBHandler().conn.read_ctx() as cursor:
+            if (value := globaldb_get_unique_cache_value(
+                cursor=cursor,
+                key_parts=(cache_type, str(vault_address)),
+            )) is not None:
+                return string_to_evm_address(value)
+
+        try:
+            if not (value := self.evm_inquirer.call_contract(
+                contract_address=vault_address,
+                abi=CURVE_VAULT_ABI,
+                method_name=contract_method,
+            )):
+                log.error(f'Empty response from {contract_method} method on Curve vault contract {vault_address}.')  # noqa: E501
+                return None
+
+            # Use to_checksum_address to fix lower case addresses
+            value = to_checksum_address(value)
+        except RemoteError as e:
+            log.error(f'Failed to call the {contract_method} method on Curve vault contract {vault_address} due to {e!s}')  # noqa: E501
+            return None
+
+        with GlobalDBHandler().conn.write_ctx() as write_cursor:
+            globaldb_set_unique_cache_value(
+                write_cursor=write_cursor,
+                key_parts=(cache_type, str(vault_address)),
+                value=value,
+            )
+        return value
+
+    def _get_possibly_cached_token(
+            self,
+            cache_type: Literal[
+                CacheType.CURVE_VAULT_BORROWED_TOKEN,
+                CacheType.CURVE_VAULT_COLLATERAL_TOKEN,
+            ],
+            vault_address: 'ChecksumEvmAddress',
+            contract_method: str,
+    ) -> 'EvmToken | None':
+        """Get token from the cache or the specified vault contract method.
+        Returns the token or None on error."""
+        if (token_address := self._get_possibly_cached_vault_address(
+            cache_type=cache_type,
+            vault_address=vault_address,
+            contract_method=contract_method,
+        )) is None:
+            return None
+
+        try:
+            return self.base.get_or_create_evm_token(
+                address=string_to_evm_address(token_address),
+            )
+        except (NotERC20Conformant, NotERC721Conformant) as e:
+            log.error(
+                f'Failed to get or create token {token_address} associated with '
+                f'Curve vault {vault_address} on {self.evm_inquirer.chain_id} due to {e}',
```

Comment:
Please use chain name. Not the chain id number for logs. Here and elsewhere

```suggestion
                f'Curve vault {vault_address} on {self.evm_inquirer.chain_name due to {e}',
```